specification/quantumleap.yml

```diff
@@ -0,0 +1,500 @@
+
+################################################################################
+# PATHS: META
+################################################################################
+paths:
+  /version:
+    get:
+      operationId: reporter.reporter.version
+      summary: "Returns the version of QuantumLeap."
+      tags:
+        - meta
+      responses:
+        200:
+          description: "Successful response."
+          schema:
+            type: object
+            properties:
+              version:
+                type: string
+          examples:
+            application/json:
+              {
+                "version": "0.0.1"
+              }
+
+  /config:
+    post:
+      operationId: reporter.reporter.config
+      summary: "Customize your persistance configuration to better suit your needs."
+      description: ""
+      parameters:
+        - $ref: '#/parameters/type'
+        - in: query
+          name: replicas
+          type: number
+          description: "The number of replicas to use for the selected types."
+      tags:
+        - meta
+      responses:
+        201:
+          description: "Successfully created record."
+
+################################################################################
+# PATHS: INPUT
+################################################################################
+  /notify:
+    post:
+      operationId: reporter.reporter.notify
+      summary: "Notify QuantumLeap the arrival of a new NGSI notification."
+      description: ""
+      parameters:
+        - in: body
+          name: notification
+          required: true
+          schema:
+            type: object
+            $ref: "#/definitions/Notification"
+      tags:
+        - input
+      responses:
+        201:
+          description: "Successfully created record."
+        400:
+          description: "Received notification is not valid."
+        500:
+          description: "Internal server error."
+
+################################################################################
+# PATHS: OUTPUT
+################################################################################
+  /entities/{entityId}/attrs/{attrName}:
+    get:
+      operationId: reporter.reporter.query_1T1E1A
+      summary: "History of an attribute of a given entity instance."
+      description: "For example, query max water level of the central tank throughout the last year. Queries can get more sophisticated with the use of filters and query attributes."
+      tags:
+        - queries
+      parameters:
+        - $ref: '#/parameters/entityIdInPath'
+        - $ref: '#/parameters/attrNameInPath'
+        # Query attrs to be refined...
+        - $ref: '#/parameters/type'
+        - $ref: '#/parameters/aggrMethod'
+        - $ref: '#/parameters/aggrPeriod'
+        - $ref: '#/parameters/options'
+        - $ref: '#/parameters/from'
+        - $ref: '#/parameters/to'
+        - $ref: '#/parameters/lastN'
+        - $ref: '#/parameters/limit'
```

Comment:
offset?

Reply:
Right, `offset` is missing.
Just to recap (for other reviewers also), using `offset` + `limit` would be like using `from`+`to`, just that the former pair works on indexes (positional numbers in response) and the latest with data (like from 01/01/2017 to today).
`lastN` will be later applied to any query result before being returned, it's like a final slicing.

Reply:
(y)